src/CADL.Extension/Emitter.Csharp/src/emitter.ts

```diff
@@ -67,88 +67,135 @@ import { OperationPaging } from "./type/OperationPaging.js";
 import { OperationLongRunning } from "./type/OperationLongRunning.js";
 import { OperationFinalStateVia } from "./type/OperationFinalStateVia.js";
 import { getOperationLink } from "@azure-tools/cadl-azure-core";
+import fs from "fs";
+import path from "node:path";
+import { Configuration } from "./type/Configuration.js";
 import { dllFilePath } from "@autorest/csharp";
 import { exec } from "child_process";
 
 export interface NetEmitterOptions {
+    "sdk-folder": string;
     outputFile: string;
     logFile: string;
+    namespace?: string;
+    "library-name"?: string;
+    "shared-source-folders"?: string[];
+    "single-top-level-client"?: boolean;
     skipSDKGeneration: boolean;
     newProject: boolean;
     configurationPath: string;
 }
 
 const defaultOptions = {
+    "sdk-folder": ".",
     outputFile: "cadl.json",
     logFile: "log.json",
     skipSDKGeneration: false,
+    "shared-source-folders": [
+        resolvePath(dllFilePath, "..", "Generator.Shared"),
+        resolvePath(dllFilePath, "..", "Azure.Core.Shared")
+    ],
     newProject: false,
     configurationPath: null
 };
 
-const EmitterOptionsSchema: JSONSchemaType<NetEmitterOptions> = {
+const NetEmitterOptionsSchema: JSONSchemaType<NetEmitterOptions> = {
     type: "object",
     additionalProperties: false,
     properties: {
+        "sdk-folder": { type: "string", nullable: true },
         outputFile: { type: "string", nullable: true },
         logFile: { type: "string", nullable: true },
+        namespace: { type: "string", nullable: true },
+        "library-name": { type: "string", nullable: true },
+        "shared-source-folders": {
+            type: "array",
+            items: { type: "string" },
+            nullable: true
+        },
+        "single-top-level-client": { type: "boolean", nullable: true },
         skipSDKGeneration: { type: "boolean", nullable: true },
         newProject: { type: "boolean", nullable: true },
         configurationPath: { type: "string", nullable: true }
     },
-    required: [],
+    required: []
 };
 
 export const $lib = createCadlLibrary({
-    name: "CSharpEmitter",
+    name: "cadl-csharp",
     diagnostics: {},
     emitter: {
-        options: EmitterOptionsSchema,
-    },
+        options: NetEmitterOptionsSchema
+    }
 });
 
 export async function $onEmit(
     program: Program,
     emitterOptions: NetEmitterOptions
 ) {
     const resolvedOptions = { ...defaultOptions, ...emitterOptions };
+    const resolvedSharedFolders: string[] = [];
+    const outputFolder = resolvePath(
+        program.compilerOptions.outputPath ?? "./cadl-output",
+        emitterOptions["sdk-folder"]
+    );
+    for (const sharedFolder of resolvedOptions["shared-source-folders"]) {
```

Comment:
shall we support absolute path for `shared-source-folders`?

Reply:
No. if it is to reuse later and it may not couple with one-time execution path. So it is better to use the relative path.